scripts/net/net-yaml-config.py

```diff
@@ -0,0 +1,298 @@
+#!/usr/bin/env python3
+# Copyright(c) 2024 Nordic Semiconductor
+# SPDX-License-Identifier: Apache-2.0
+
+import yaml, sys, socket
+
+data = yaml.safe_load(sys.stdin)
+
+# If pykwalify is installed, then validate the yaml
+try:
+    import pykwalify.core
+
+    def yaml_validate(data, schema):
+        if not schema:
+            return
+        c = pykwalify.core.Core(source_data=data, schema_files=[schema])
+        c.validate(raise_exception=True)
```

Comment:
This validates the input file against the schema, but not the schema against the parser.